src/client/datascience/interactive-window/interactiveWindow.ts

```diff
@@ -155,11 +142,17 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
             this._submitters.push(owner);
         }
 
+        // Request creation of the interactive window from VS Code
+        this._editorReadyPromise = this.createEditorReadyPromise();
+
         // Wait for a controller to get selected
-        this.initialControllerSelected = createDeferred<void>();
+        this._controllerReadyPromise = createDeferred<VSCodeNotebookController>();
 
-        // Request creation of the interactive window from VS Code
-        this._editorReadyPromise = this.createReadyPromise();
+        // Set up promise for kernel ready
+        this._kernelReadyPromise = this.createKernelReadyPromise();
+
+        // Immediately start trying to connect to kernel
```

Comment:
This line shouldn't be necessary AFAIK. Creation of the promise starts the whole thing.